src/lrf.rs

```diff
@@ -964,11 +966,12 @@ impl RestorationState {
       for si in 0..stripe_n {
         let (stripe_start_y, stripe_size) = if si == 0 {
           (0, (64 - 8) >> ydec)
-        }  else {
+        } else {
+          let start = (si * 64 - 8) >> ydec;
           (
-            (si as isize * 64 - 8) >> ydec,
+            start as isize,
             // one past, unlike spec
```

Comment:
I'm unsure what this comment means.

Reply:
I'm using a C-style fencepost so you can check it against foo < end unlike the spec where they use foo <= end

Reply:
The spec uses end fenceposts set to the last valid entry and checks that accesses are <= that value.
I used a C-style fencepost value that's one past the end and check that accesses are < that value.